### PR TITLE
chore(manifest): environment manifest does not require az if VPC is adjusted

### PR DESCRIPTION
**Pull request by Lou1415926**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: internal/pkg/manifest/env.go [hunk 1]
```diff
@@ -247,17 +251,20 @@ func (cfg *environmentVPCConfig) ManagedVPC() *template.ManagedVPC {
...
 		return aws.StringValue(cfg.Subnets.Private[i].AZ) < aws.StringValue(cfg.Subnets.Private[j].AZ)
 	})
 	for idx, subnet := range cfg.Subnets.Public {
 		publicSubnetCIDRs[idx] = aws.StringValue((*string)(subnet.CIDR))
 		privateSubnetCIDRs[idx] = aws.StringValue((*string)(cfg.Subnets.Private[idx].CIDR))
 		azs[idx] = aws.StringValue(subnet.AZ)
 	}
+	if azs[0] == "" {
+		azs = nil
+	}
```
Comment (iamhopaul123) [inline internal/pkg/manifest/env.go]: this looks so weird...can we just make `var azs []string`?

File: internal/pkg/manifest/validate_env.go [hunk 2]
```diff
@@ -104,28 +104,34 @@ func (cfg environmentVPCConfig) validateManagedVPC() error {
 	)
 	var exists = struct{}{}
 	for idx, subnet := range cfg.Subnets.Public {
-		if aws.StringValue((*string)(subnet.CIDR)) == "" || aws.StringValue(subnet.AZ) == "" {
-			return fmt.Errorf(`validate "subnets": public[%d] must include "cidr" and "az" fields if the vpc is configured`, idx)
+		if aws.StringValue((*string)(subnet.CIDR)) == "" {
+			return fmt.Errorf(`validate "subnets": public[%d] must include "cidr" if the vpc is configured`, idx)
```
Comment (iamhopaul123) [inline internal/pkg/manifest/validate_env.go]: can we use https://github.com/aws/copilot-cli/blob/8e6e7a17bb045bba643b2084d8b12b22fca447c8/internal/pkg/manifest/errors.go#L57 ?

File: internal/pkg/manifest/env.go (hunk 1 above)
Comment (Lou1415926) [reply internal/pkg/manifest/env.go]: updated!

File: internal/pkg/manifest/validate_env.go [hunk 3]
```diff
@@ -235,3 +241,15 @@ func (c environmentConfig) validateInternalALBSubnets() error {
 	}
 	return nil
 }
+
+func areSetsEqual[T comparable](a map[T]struct{}, b map[T]struct{}) bool {
```
Comment (efekarakus) [inline internal/pkg/manifest/validate_env.go:257]: one of these days, we should add a `sets` pkg like how we have a `graphs` pkg

**Review by efekarakus — APPROVED** [review]
:shipit: applied label in case you'd like to address Penghao's feedback

File: internal/pkg/manifest/validate_env.go (hunk 2 above)
Comment (Lou1415926) [reply internal/pkg/manifest/validate_env.go]: done!

File: internal/pkg/manifest/validate_env.go (hunk 3 above)
Comment (iamhopaul123) [reply internal/pkg/manifest/validate_env.go:257]: Why is very basic `set()` not supported in Go 😤

**Review by iamhopaul123 — APPROVED** [review]

File: internal/pkg/manifest/validate_env.go (hunk 3 above)
Comment (dannyrandall) [reply internal/pkg/manifest/validate_env.go:257]: this'll exist soon (hopefully)! https://pkg.go.dev/golang.org/x/exp/maps#Equal

**Pull request merged**